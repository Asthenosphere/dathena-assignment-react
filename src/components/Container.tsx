import React from "react";

import "./Container.scss";

interface ContainerProps {
  children: React.ReactNode;
}

// Container component that displays its children at the center of the page.
const Container: React.FC<ContainerProps> = (props: ContainerProps) => {
  return <div className='container ion-padding'>{props.children}</div>;
};

export default Container;
