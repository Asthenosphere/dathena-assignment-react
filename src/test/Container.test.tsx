import React from "react";
import Container from "../components/Container";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";

configure({ adapter: new Adapter() });

it("renders conatiner correctly", () => {
  const container = shallow(<Container>Hello</Container>);
  expect(container.getElements()).toMatchSnapshot();
});
