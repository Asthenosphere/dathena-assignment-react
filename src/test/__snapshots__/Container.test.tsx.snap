// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders conatiner correctly 1`] = `
Array [
  <div
    className="container ion-padding"
  >
    Hello
  </div>,
]
`;
